lib/mox.ex

```diff
@@ -183,6 +183,15 @@ defmodule Mox do
   `$callers` is used to determine the process that actually defined the
   expectations.
 
+  #### Explicit allowances as promises
+
+  Under some circumstances, the process might not have been already started
+  when the allowance happens. In such a case, you might specify the allowance
+  as a promise in a form `(-> pid())`. This promise would be
+  resolving late, at the very moment of dispatch. Please note, that no
+  additional diagnostics could have been provided for promises, and it would
+  either succeed, or fail with `Mox.UnexpectedCallError`. 
```

Comment:
Can you please reword to not mention "promises"? "Lazy" would be fine but we can also just use "functions". Thank you.

Reply:
Also please rename the variables and other parts of the docs accordingly.